packages/@lwc/state/src/lwc-adapter.ts

```diff
@@ -0,0 +1,64 @@
+import type { Signal } from '@lwc/signals';
+import type { LightningElement } from 'lwc';
+import type { RuntimeAdapter } from './runtime-interface.js';
+import { symbolContextKey, ContextRequestEvent } from './event.js';
+
+export class LWCAdapter implements RuntimeAdapter<LightningElement> {
```

Comment:
Might be worth adding a note to this file that the functionality will be migrated into the LWC framework itself.